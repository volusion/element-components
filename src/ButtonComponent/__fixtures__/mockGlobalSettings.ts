import { GlobalSettings } from '../types';
export const mockGlobalSettings: GlobalSettings = {
    globalComponents: {
        volComponentButton: {
            primaryButtonStyles: {
                size: 'large',
                rounded: 'none',
                textColor: '#fff',
                fontWeight: '700',
                borderColor: 'rgba(247, 67, 186, 1)',
                growOnHover: true,
                letterSpacing: 'none',
                textTransform: 'none',
                hoverTextColor: '#fff',
                backgroundColor: 'rgba(247, 68, 186, 1)',
                borderThickness: 'none',
                hoverBorderColor: 'rgba(0, 0, 0, 1)',
                hoverBackgroundColor: 'rgba(218, 15, 148, 1)'
            },
            secondaryButtonStyles: {
                size: 'block',
                rounded: 'none',
                textColor: 'rgba(255, 255, 255, 1)',
                fontWeight: '700',
                borderColor: 'rgba(247, 67, 186, 1)',
                growOnHover: false,
                letterSpacing: 'none',
                textTransform: 'none',
                hoverTextColor: 'rgba(255, 255, 255, 1)',
                backgroundColor: 'rgba(247, 74, 201, 1)',
                borderThickness: 'thin',
                hoverBorderColor: 'rgba(0, 0, 0, 1)',
                hoverBackgroundColor: 'rgba(0, 0, 0, 1)'
            }
        }
    }
};
