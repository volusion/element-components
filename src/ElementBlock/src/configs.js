import { ElementPropTypes } from '@volusion/element-proptypes';

export const configSchema = {
    text: {
        label: 'Text content',
        type: ElementPropTypes.string
    }
};

export const defaultConfig = {
    text: 'Element Starter Block'
};
