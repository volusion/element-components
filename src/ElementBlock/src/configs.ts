import { ElementBlockConfigs } from './types';
import { ElementPropTypes } from '@volusion/element-proptypes';

export const configSchema = {
    isFullWidth: {
        label: `Full width`,
        type: ElementPropTypes.bool
    },
    hasHorizontalPadding: {
        label: `Horizontal spacing`,
        type: ElementPropTypes.bool
    },
    topPadding: {
        label: `Top spacing`,
        type: ElementPropTypes.oneOf(['None', 'Small', 'Medium', 'Large'])
    },
    bottomPadding: {
        label: `Top spacing`,
        type: ElementPropTypes.oneOf(['None', 'Small', 'Medium', 'Large'])
    },
    showOnDesktop: {
        label: `Show on desktop`,
        type: ElementPropTypes.bool
    },
    showOnTablet: {
        label: `Show on tablet`,
        type: ElementPropTypes.bool
    },
    showOnMobile: {
        label: `Show on mobile`,
        type: ElementPropTypes.bool
    },
    customAttrs: {
        label: `Custom element attributes`,
        type: ElementPropTypes.arrayOf(
            ElementPropTypes.shape({
                name: {
                    label: `Name`,
                    type: ElementPropTypes.string
                },
                value: {
                    label: `Value`,
                    type: ElementPropTypes.string
                }
            })
        )
    }
};

export const defaultConfig: ElementBlockConfigs = {
    isFullWidth: true,
    hasHorizontalPadding: false,
    topPadding: 'None',
    bottomPadding: 'None',
    showOnDesktop: true,
    showOnTablet: true,
    showOnMobile: true,
    customAttrs: [
        {
            name: '',
            value: ''
        }
    ]
};
