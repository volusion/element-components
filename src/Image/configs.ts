import { ElementPropTypes } from '@volusion/element-proptypes';

export const defaultConfig = {
    alt: '',
    height: 270,
    src:
        'https://res.cloudinary.com/dyx4yhvoq/image/upload/v1545428185/images/tcp-no-image.jpg',
    title: '',
    width: 360
};

export const configSchema = {
    src: {
        label: 'Image',
        type: ElementPropTypes.string
    },
    title: {
        label: 'Title',
        type: ElementPropTypes.string
    },
    altText: {
        label: 'Alt Text',
        type: ElementPropTypes.string
    },
    width: {
        label: 'Width',
        type: ElementPropTypes.number
    },
    height: {
        label: 'Height',
        type: ElementPropTypes.number
    }
};
