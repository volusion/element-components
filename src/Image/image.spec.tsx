import React from 'react';
import { mount, render } from 'enzyme';
import { block as Image, ImageProps } from './index';
import { defaultConfig } from './configs';
import { mockGlobalSettings as globalSettings } from '../__fixtures__/mockGlobalSettings';

describe('Image component', () => {
    let props: ImageProps;

    beforeEach(() => {
        props = {
            ...defaultConfig,
            globalSettings,
            utils: {
                isAmpRequest: false
            }
        };
    });

    it('matches snapshot', () => {
        const rendered = render(<Image {...props} />);
        expect(rendered).toMatchSnapshot();
    });

    it('should have an img tag when a normal request', () => {
        const image = mount(<Image {...props} />);
        expect(image.find('img').exists()).toBeTruthy();
    });

    it('should have an amp-img tag when an AMP request', () => {
        props.utils.isAmpRequest = true;
        const image = mount(<Image {...props} />);
        expect(image.find('img').exists()).toBeFalsy();
        expect(image.find('amp-img').exists()).toBeTruthy();
        expect(image.render()).toMatchSnapshot();
    });

    it('should change src, alt text, and title when given props', () => {
        props.alt = 'fake alt text';
        props.title = 'fake title';
        props.src = 'https://www.fakeimgsrc.com/fake/image.jpg';
        const image = mount(<Image {...props} />);
        expect(image.find('img').prop('alt')).toEqual(props.alt);
        expect(image.find('img').prop('title')).toEqual(props.title);
        expect(image.find('img').prop('src')).toEqual(props.src);
        expect(image.render()).toMatchSnapshot();
    });
});
