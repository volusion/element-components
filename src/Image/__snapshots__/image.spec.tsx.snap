// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component matches snapshot 1`] = `
<picture>
  <img
    alt=""
    height="270"
    src="https://res.cloudinary.com/dyx4yhvoq/image/upload/v1545428185/images/tcp-no-image.jpg"
    style="max-width:100%"
    title=""
    width="360"
  />
</picture>
`;

exports[`Image component should change src, alt text, and title when given props 1`] = `
<picture>
  <img
    alt="fake alt text"
    height="270"
    src="https://www.fakeimgsrc.com/fake/image.jpg"
    style="max-width: 100%;"
    title="fake title"
    width="360"
  />
</picture>
`;

exports[`Image component should have an amp-img tag when an AMP request 1`] = `
<amp-img
  alt=""
  height="270"
  layout="responsive"
  src="https://res.cloudinary.com/dyx4yhvoq/image/upload/v1545428185/images/tcp-no-image.jpg"
  title=""
  width="360"
/>
`;
